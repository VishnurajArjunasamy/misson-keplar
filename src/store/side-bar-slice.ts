import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import { SideBarState } from "../modals/sidebar-modal";
import { getMemebers } from "../services/members";
import { ERROR_MSG } from "../constants/app.constants";

export const fetchMembers = createAsyncThunk(
  "members/fetchMembers",
  async () => {
    return getMemebers();
  }
);

const inititalState: SideBarState = {
  showMembers: false,
  isDarkMode: false,
  members: null,
  loading: false,
  error: null,
};

export const sideBarSlice = createSlice({
  name: "SideBar",
  initialState: inititalState,
  reducers: {
    toggleShowMembers: (state) => {
      state.showMembers = !state.showMembers;
    },
    toggleDarkMode: (state) => {
      state.isDarkMode = !state.isDarkMode;
    },
    setMembers: (state, action) => {
      state.members = action.payload;
    },
    setLoading: (state, action) => {
      state.loading = action.payload;
    },
    setError: (state, action) => {
      state.error = action.payload;
    },
  },
  extraReducers: (builder) => {
    builder
      .addCase(fetchMembers.pending, (state) => {
        state.loading = true;
      })
      .addCase(fetchMembers.fulfilled, (state, action) => {
        state.loading = false;
        state.members = action.payload;
      })
      .addCase(fetchMembers.rejected, (state, action) => {
        (state.loading = false),
          (state.error = action.error.message || ERROR_MSG.MESSAGE);
      });
  },
});

export const {
  toggleDarkMode,
  toggleShowMembers,
  setMembers,
  setLoading,
  setError,
} = sideBarSlice.actions;

export default sideBarSlice.reducer;
