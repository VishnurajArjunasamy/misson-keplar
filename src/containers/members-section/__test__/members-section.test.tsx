import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import MembersSection from "../members-section";
import { fetchMembers } from "../../../store/side-bar-slice";
import "@testing-library/jest-dom";

// Create a mock store
const mockStore = configureMockStore();

jest.mock("../../../store/side-bar-slice", () => ({
  fetchMembers: jest.fn(),
}));

describe("MemebersSection Component", () => {
  let store: ReturnType<typeof mockStore>;

  beforeEach(() => {
    store = mockStore({
      sideBar: {
        members: [],
        loading: false,
        error: null,
        isDarkMode: false,
      },
    });
    store.dispatch = jest.fn();
  });

  it("dispatches fetchMembers on mount", () => {
    render(
      <Provider store={store}>
        <MembersSection />
      </Provider>
    );

    expect(store.dispatch).toHaveBeenCalledWith(fetchMembers());
  });
});
