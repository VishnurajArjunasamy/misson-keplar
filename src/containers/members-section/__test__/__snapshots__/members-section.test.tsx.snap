// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemebersSection Component matches snapshot 1`] = `[Function]`;
